webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx

```diff
@@ -1,283 +1,269 @@
+const MarketplaceModal = ({
+    openedFrom,
+}: MarketplaceModalProps) => {
+    const dispatch = useDispatch();
+    const {formatMessage} = useIntl();
+    const listRef = useRef<HTMLDivElement>(null);
+
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.PLUGIN_MARKETPLACE));
+    const listing = useSelector(getListing);
+    const installedListing = useSelector(getInstalledListing);
+    const pluginStatuses = useSelector((state: GlobalState) => state.entities.admin.pluginStatuses);
+    const firstAdminVisitMarketplaceStatus = useSelector(getFirstAdminVisitMarketplaceStatus);
+
+    const [tabKey, setTabKey] = useState(MarketplaceTabs.ALL_LISTING);
+    const [filter, setFilter] = useState('');
+    const [page, setPage] = useState(0);
+    const [hasLoaded, setHasLoaded] = useState(false);
+    const [loading, setLoading] = React.useState(true);
+    const [serverError, setServerError] = React.useState(false);
+
+    const doFetchListing = useCallback(async () => {
+        const {error} = await dispatch(fetchListing()) as ActionResult;
+
+        if (error) {
+            setServerError(true);
+        }
 
-        this.filterRef = React.createRef();
-    }
+        setLoading(false);
+    }, []);
 
-    componentDidMount(): void {
-        trackEvent('plugins', 'ui_marketplace_opened', {from: this.props.openedFrom});
+    const doSearch = useCallback(async () => {
+        trackEvent('plugins', 'ui_marketplace_search', {filter});
 
-        this.fetchListing();
-        this.props.actions.getPluginStatuses();
-        if (!this.props.firstAdminVisitMarketplaceStatus) {
-            trackEvent('plugins', 'ui_first_admin_visit_marketplace_status');
+        const {error} = await dispatch(filterListing(filter)) as ActionResult;
 
-            this.props.actions.setFirstAdminVisitMarketplaceStatus();
+        if (error) {
+            setServerError(true);
         }
+    }, [filter]);
 
-        this.filterRef.current?.focus();
-    }
+    const debouncedSearch = debounce(doSearch, SEARCH_TIMEOUT_MILLISECONDS);
 
-    componentDidUpdate(prevProps: MarketplaceModalProps): void {
-        // Automatically refresh the component when a plugin is installed or uninstalled.
-        if (this.props.pluginStatuses !== prevProps.pluginStatuses) {
-            this.fetchListing();
+    useEffect(() => {
+        async function doFetch() {
+            await dispatch(getPluginStatuses());
+            await doFetchListing();
+            setHasLoaded(true);
         }
-    }
 
-    fetchListing = async (): Promise<void> => {
-        const {error} = await this.props.actions.fetchListing();
-        this.setState({loading: false, serverError: error});
-    }
+        trackEvent('plugins', 'ui_marketplace_opened', {from: openedFrom});
 
-    close = (): void => {
-        trackEvent('plugins', 'ui_marketplace_closed');
-        this.props.actions.closeModal();
-    }
+        if (firstAdminVisitMarketplaceStatus) {
```

Comment:
Noticed that before this was `if (!this.props.firstAdminVisitMarketplaceStatus)` which seems it was a bug and you are fixing it here removing the not condition?

Reply:
Nice catch! I was tricked by the config name. Updated along with the const name to make it more clear.

Reply:
Good renaming - it was also confusing for me when I first saw it and thought the logic was wrong. 

We need to test this case of the tracking being emitted only once for admins. @julmondragon could you include this in the QA steps to be sure we cover it? cc/ @stevemudie